docs/taskruns.md

```diff
@@ -236,82 +236,29 @@ spec:
-- [`secret`](https://kubernetes.io/docs/concepts/storage/volumes/#secret)
-
-_If you need support for a `VolumeSource` not listed here
-[please open an issue](https://github.com/tektoncd/pipeline/issues) or feel free to
-[contribute a PR](https://github.com/tektoncd/pipeline/blob/master/CONTRIBUTING.md)._
-
-If the declared `workspaces` are not provided at runtime, the `TaskRun` will fail
-with an error.
-
-For example to provide an existing PVC called `mypvc` for a `workspace` called
-`myworkspace` declared by the `Task`, using the `my-subdir` folder which already exists
-on the PVC (there will be an error if it does not exist):
+Here are the relevant fields of a `TaskRun` spec when providing a
+`PersistentVolumeClaim` as a workspace:
 
 ```yaml
 workspaces:
-- name: myworkspace
+- name: myworkspace # must match workspace name in Task
   persistentVolumeClaim:
-    claimName: mypvc
+    claimName: mypvc # this PVC must already exist
   subPath: my-subdir
 ```
 
-Or to use [`emptyDir`](https://kubernetes.io/docs/concepts/storage/volumes/#emptydir) for the same `workspace`:
-
-```yaml
-workspaces:
-- name: myworkspace
-  emptyDir: {}
-```
-
-A ConfigMap can also be used as a workspace with the following caveats:
-
-1. ConfigMap volume sources are always mounted as read-only inside a task's
-containers - tasks cannot write content to them and a step may error out
-and fail the task if a write is attempted.
-2. The ConfigMap you want to use as a workspace must already exist prior
-to the TaskRun being submitted.
-3. ConfigMaps have a [size limit of 1MB](https://github.com/kubernetes/kubernetes/blob/f16bfb069a22241a5501f6fe530f5d4e2a82cf0e/pkg/apis/core/validation/validation.go#L5042)
-
-To use a [`configMap`](https://kubernetes.io/docs/concepts/storage/volumes/#configmap)
-as a `workspace`:
-
-```yaml
-workspaces:
-- name: myworkspace
-  configmap:
-    name: my-configmap
-```
-
-A Secret can also be used as a workspace with the following caveats:
+For more examples and complete documentation on configuring `workspaces` in
+`TaskRun`s see [workspaces.md](./workspaces.md#providing-workspaces-with-taskruns).
 
-1. Secret volume sources are always mounted as read-only inside a task's
-containers - tasks cannot write content to them and a step may error out
-and fail the task if a write is attempted.
-2. The Secret you want to use as a workspace must already exist prior
-to the TaskRun being submitted.
-3. Secrets have a [size limit of 1MB](https://github.com/kubernetes/kubernetes/blob/f16bfb069a22241a5501f6fe530f5d4e2a82cf0e/pkg/apis/core/validation/validation.go#L4933)
-
-To use a [`secret`](https://kubernetes.io/docs/concepts/storage/volumes/#secret)
-as a `workspace`:
-
-```yaml
-workspaces:
-- name: myworkspace
-  secret:
-    secretName: my-secret
-```
+Tekton supports several different kinds of `Volume` in `Workspaces`. For a list of
+the different kinds see the section of `workspaces.md` on
```

Comment:
Leave out the workspaces.md filename. Just say "For a list of hte different kinds, see [`VolumeSources...`](restofyourlinkhere).